lms/djangoapps/certificates/models.py

```diff
@@ -842,26 +842,44 @@ def course_key(self):
 class CertificateGenerationCourseSetting(TimeStampedModel):
     """Enable or disable certificate generation for a particular course.
 
-    This controls whether students are allowed to "self-generate"
-    certificates for a course.  It does NOT prevent us from
-    batch-generating certificates for a course using management
-    commands.
-
     In general, we should only enable self-generated certificates
     for a course once we successfully generate example certificates
     for the course.  This is enforced in the UI layer, but
     not in the data layer.
-
     """
     course_key = CourseKeyField(max_length=255, db_index=True)
-    enabled = models.BooleanField(default=False)
+    enabled = models.BooleanField(
+        default=False,
+        help_text=_(
+            u"DEPRECATED, please use 'self generation enabled' instead."
+        )
+    )
+    # TODO: Learner-2549 remove deprecated enabled field
+
+    self_generation_enabled = models.BooleanField(
+        default=False,
+        help_text=_(
+            u"controls whether students are allowed to 'self-generate' "
```

Comment:
Allow students to generate their own certificates for the course. Enabling this does NOT affect usage of the management command used for batch certificate generation.